### PR TITLE
ease testsetup for JobContextService

### PR DESCRIPTION
**Pull request by mfelsche**

### CODE REVIEW

File: sql/src/main/java/io/crate/jobs/JobContextService.java [hunk 1]
```diff
@@ -36,26 +37,34 @@
...
+    @BindingAnnotation
+    @Target({ ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD })
+    @Retention(RetentionPolicy.RUNTIME)
+    public static @interface ReaperInterval {}
 
 
     private final ThreadPool threadPool;
     private final StatsTables statsTables;
-    private final ScheduledFuture<?> keepAliveReaper;
+    private ScheduledFuture<?> keepAliveReaper;
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/jobs/JobContextService.java]: why did you remove the final?

File: sql/src/main/java/io/crate/jobs/JobContextService.java [hunk 2]
```diff
@@ -63,16 +72,29 @@
     private final ReentrantReadWriteLock.ReadLock readLock = rwLock.readLock();
     private final ReentrantReadWriteLock.WriteLock writeLock = rwLock.writeLock();
 
+    private final TimeValue keepAlive;
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/jobs/JobContextService.java]: can also be declared as final

File: sql/src/main/java/io/crate/jobs/Reaper.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package io.crate.jobs;
+
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.util.Collection;
+
+public interface Reaper {
+    public void killHangingJobs(TimeValue keepAlive, Collection<JobExecutionContext> contexts);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/jobs/Reaper.java]: is keepAlive the interval? Either rename to something more specific or add javadocs

File: sql/src/main/java/io/crate/jobs/KillJobsReaper.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+@Singleton
+public class KillJobsReaper implements Reaper {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/jobs/KillJobsReaper.java]: remove as it is unused

File: sql/src/main/java/io/crate/jobs/JobContextService.java (hunk 2 above)
Comment (mfelsche) [inline sql/src/main/java/io/crate/jobs/JobContextService.java]: it is

**Comment by mfussenegger** [comment]
ack
